bluepysnap/circuit_validation.py

```diff
@@ -408,6 +415,53 @@ def _check(indices, nodes_ds):
     return errors
 
 
+def _get_edge_population_groups(population):
+    """Get groups from a edge population."""
+    return [name for name in population
+            if name != "indices" and isinstance(population[name], h5py.Group)]
+
+
+def _check_edge_group_id(population):
+    errors = []
+    groups = np.array(_get_edge_population_groups(population))
+    group_datasets = ["edge_group_id", "edge_group_index"]
+    missing_datasets = set(group_datasets) - set(population)
+    population_name = _get_group_name(population)
+    if len(groups) > 1:
+        errors.append(BbpError(Error.WARNING, 'Population {} of {} have multiple groups. '
+                                              'Cannot be read via bluepysnap or libsonata'.
+                               format(population_name, population.file.filename)))
+    if len(missing_datasets) == 2 and len(groups) == 1:
+        # no "edge_group_id", "edge_group_index" and only one group --> can use implicit ids
+        return errors
+    elif len(missing_datasets) == 1:
+        return errors + [fatal('Population {} of {} misses dataset {}'.
+                               format(population_name, population.file.filename, missing_datasets))]
+
+    edge_group_ids = population["edge_group_id"][:]
```

Comment:
Will we get here in case of `len(missing_datasets) == 2 and len(groups) > 1` ?

Reply:
no you should raise at line 435

Reply:
Will we get into it if `len(groups) > 1`. The condition is for `len(groups)==1`.

Reply:
we want to go there if (group >= 1 and len(missing_group_datasets)==0)

Reply:
if  (len(groups)>1 and len(missing_group_datasets)>0) I raise at `if missing_datasets:`  line 440